Automattic/wp-calypso
# Use DataViews for all domains list

Related to #96674 

This solution doesn't address one main point we wanted to have in the beginning and also received as feedback: we don't have all the needed data immediately from the backend.

It is a trade-off for the first iteration.

Implementation of the needed backend has its challenges and will be addressed later: pdxWSz-23k-p2

## Proposed Changes

* Introduce DataViews for all domains list.
* Add `calypso/domains-dataviews` feature flag.

## Why are these changes being made?

* Use the standard approach to render a list on WordPress.

## Testing Instructions

* Turn on feature flags in your browser's console: `window.sessionStorage.setItem('flags',  'calypso/domains-dataviews'])`
* Go to http://calypso.localhost:3000/domains/manage
* Check filtering, sorting, actions, bulk actions.
    * For bulk actions, you need to have at least two domains you own. 
* Click on a domain and check if it opened a pane on the right.
* Switch the tab on the right to Email.
* Click on another domain and make sure the same tab (Email) opened for it.
* Now disable DataViews: `window.sessionStorage.removeItem('flags');` and test everything works with the previous implementation.

## Pre-merge Checklist
